docker/README.md

```diff
@@ -1 +1,112 @@
+    |                          |            
+    + - - - -  - + - - - - - - +            
+                 |
+        + ---------------- +
+        | Bitcoin network  |  <---  In current scenarion for simplicity we create only one  
+        + ---------------- +        "btcd" node which represents the Bitcoin network, in  
+                                    real situation Alice and Bob will likely be 
+                                    connected to different Bitcoin nodes.
+```
+
+#### Prerequisites
+    Name  | Version 
+  --------|---------
+  docker-compose | 1.9.0
+  docker | 1.13.0
+
+**General workflow is following:** 
+ * Create Bitcoin network
+ * Create `Alice` node
+ * Create `Bob` node
+ * Initialize `Alice` node with some amount of bitcoins
+ * Open channel between `Alice` and `Bob`
+ * Send payment from `Alice` to `Bob`
+
+Start the `btcd`, create `Alice` address and mine some bitcoins.
+```bash
+# Create "btcd" node:
+$ docker-compose up -d "btcd"
+
+# Run "Alice" container and log into it:
+$ docker-compose up -d "alice"
+$ docker exec -i -t "alice" bash
+
+# Generate new backward compatible "segwit" "alice" address:
+alice$ lncli newaddress np2wkh 
+
+# Recreate "btcd" node and set "alice" address as mining address:
+$ MINING_ADDRESS=<alice_address> docker-compose up -d "btcd"
+
+# Generate 120 block (we need at least "100 >=" blocks because of coinbase 
+# block maturity and "201 >=" in order to activate segwit):
+$ docker-compose run btcctl generate 201
+```
+
+Now we have `btcd` node turned on and some amount of bitcoins mined on 
+`Alice` address. But in order `lnd` saw them lets restart the `lnd` 
+daemon.
+```bash
+# Stop "Alice" container:
+$ docker-compose stop "alice"
+
+# Start "Alice" container and log into it:
+$ docker-compose up --no-recreate -d "alice"
+$ docker exec -i -t "alice" bash
+
+# Check "Alice" balance:
+alice$ lncli walletbalance --witness_only=true
+```
+
+Connect `Bob` node to `Alice` node.
+```bash
+# Run "Bob" node and log into it:
+$ docker-compose up --no-recreate -d "bob"
+$ docker exec -i -t "bob" bash
+
+# Get the identity address of "Bob" node:
+bob$ lncli getinfo
+
+# Get the IP address of "Bob" node:
+$ docker inspect "bob" | grep IPAddress
+
+# Connect "Alice" to the "Bob" node:
+alice$ lncli connect <bob_identity_address>@<bob_host>:10011
+
+# Check list of peers:
+alice$ lncli listpeers
+```
+
+Create `Alice<->Bob` channel.
+```bash
+# Open the channel with "Bob":
+alice$ lncli openchannel --node_key="<bob_lightning_id>" --num_confs=1 --local_amt=1000000 --push_amt=0
+
+# Include funding transaction in block thereby open the channel:
+$ docker-compose run btcctl generate 1
+
+# Check that channel with "Bob" was created:
+alice$ lncli listchannels
+```
+
+Send the payment form "Alice" to "Bob".
+```
+# Add invoice on "Bob" side:
+bob> lncli addinvoice --value=10000 --preimage="0000000000000000000000000000000000000000000000000000000000000000"
```

Comment:
You can leave off the `preimage` parameters, and the daemon is create one with a random payment hash.